src/ppx/xr_component.cpp

```diff
@@ -733,30 +733,39 @@ void XrComponent::EndFrame(const std::vector<grfx::SwapchainPtr>& swapchains, ui
 void XrComponent::ConditionallyPopulateProjectionLayer(const std::vector<grfx::SwapchainPtr>& swapchains, uint32_t startIndex, XrLayerBaseQueue& layerQueue, XrProjectionLayer& projectionLayer)
 {
     const size_t viewCount = mViews.size();
-    PPX_ASSERT_MSG(swapchains.size() >= (viewCount + startIndex), "Number of swapchains needs to be larger than or equal to the number of views!");
+    PPX_ASSERT_MSG(IsMultiView() || swapchains.size() >= (viewCount + startIndex), "Number of swapchains needs to be larger than or equal to the number of views!");
 
     if (!mShouldRender) {
         return;
     }
 
+    // In Multiview we have one shared swapchain, but the array index moves
+    // Non Multiview swapchain per view, but array index stays still.
+    const int indexToSwapchainMultiplier  = (IsMultiView() ? 0 : 1);
+    const int indexToArrayIndexMultiplier = (IsMultiView() ? 1 : 0);
+
     // Projection and (optional) depth info layer from color+depth swapchains.
     for (size_t i = 0; i < viewCount; ++i) {
+        const uint32_t swapchain_index = startIndex + (indexToSwapchainMultiplier * i);
+
         XrCompositionLayerProjectionView view = {XR_TYPE_COMPOSITION_LAYER_PROJECTION_VIEW};
         view.pose                             = mViews[i].pose;
         view.fov                              = mViews[i].fov;
-        view.subImage.swapchain               = swapchains[startIndex + i]->GetXrColorSwapchain();
+        view.subImage.swapchain               = swapchains[swapchain_index]->GetXrColorSwapchain();
+        view.subImage.imageArrayIndex         = i * indexToArrayIndexMultiplier;
```

Comment:
personal preference:
I'd be in favor or not using `indexToSwapchainMultipler` or similar, but something like:

```cpp
const bool isMultiView = isMultiView();
for (...)
    swapchain_index = isMultiView ? startIndex : startIndex + i;
    [...]
    view.subImage.imageArrayIndex = isMultiView ? i : 0;
```